server/core/terraform/terraform_client.go

```diff
@@ -237,7 +243,10 @@ func (c *DefaultClient) RunCommandWithVersion(ctx context.Context, prjCtx comman
 				err = line.Err
 				break
 			}
-			lines = append(lines, line.Line)
+			// sanitize output by stripping out logs matching config filter
+			if !c.logFilter.LogLineShouldBeFiltered(line.Line) {
+				lines = append(lines, line.Line)
+			}
 		}
 		output := strings.Join(lines, "\n")
 
```

Comment:
Offline, we discussed moving the log filter over to the UI output code layer, allowing for the original logs to be persisted in storage layer. For the GH UI, I just wanted to suggest a compromise of placing the filter here, which achieves the original goal, and prevents us from having to inject the filter in more places since the runners parse the output in several different ways. It felt a little cleaner to move it here, and have it exist alongside the ansi char filter as well.

Reply:
why would it have to be in multiple places? The only place we write github output is in the markdown renderer which should be self contained.

I don't think mirroring the ansi filter is a reasonable justification since arguably it's not the right place for that either imo.  The ansi filter should just be another filter that's hidden by the abstraction you've created.